rule-templates/ruby/base-rules.mdc

```diff
@@ -1,5 +1,256 @@
+
+### Code Quality Anti-patterns to Avoid
+- **Fat Models:** Models with too much business logic - move to service objects or POROs
+- **God Classes:** Classes doing too much - break into smaller, focused classes
+- **Duplicated Code:** Repeated code - follow DRY principle and extract to shared methods
+- **Long Methods:** Complex methods - break into smaller, focused methods
+- **Magic Numbers:** Hardcoded values - use constants or enums instead
+
+## API Conventions
+- Use versioning in URLs: `/api/v1/resource`
+- Maintain backward compatibility between versions
+- Use appropriate HTTP status codes
+- Keep consistent response format
+- Include relevant metadata in responses
+- Write tests for all API endpoints
+- Maintain high code coverage
+- Avoid using callbacks in models
+- Use namespaces for business areas (e.g., "payments")
+
+## Authentication & Authorization
+- Implement JWT authentication
+- Use Pundit for authorization policies
+- Keep authorization logic in policies
+- Each model should have its corresponding policy
+
+## Performance Optimization
+- Use database indexing effectively
+- Use ActiveRecord methods instead of raw SQL when possible
+- Avoid N+1 queries with `includes`, `preload`, or `eager_load`
+- Optimize queries using includes, joins, or select
+
+## Code Style and Documentation
+### Comments Guidelines
+- Write comments only when absolutely necessary
+- All comments must be in English
+- Use comments for complex business logic or non-obvious code
+
+### Documentation Standards
+- Keep project documentation up to date
+- Document all public APIs and interfaces
+- Update documentation with significant changes
+- Use clear and concise language
+
+### Language Requirements
+- All code-related communication in English
+- All AI assistant responses in Spanish
+
+### Security Practices
+- Follow security best practices
+- Follow principle of least privilege
+- Use environment variables for configuration
+
+## Development Workflow
+### Rails Generators
+Use appropriate generators:
+- `rails generate model` for new models
+- `rails generate migration` for database changes
+- `rails generate controller` for new controllers
+- `rails generate service` for new services
+- `rails generate policy` for new policies
+- `rails generate serializer` for new serializers
+- `rails generate job` for new background jobs
+- `rails generate form` for new form objects
+- `rails generate decorator` for new decorators
+
+### Setup Commands
+- Check Ruby version in `.ruby-version` before starting
+- Review `Gemfile` for required dependencies and gem versions
+- Run `bundle install` after Gemfile changes
+- Use `rails db:migrate` after creating migrations
+- Run `rails db:seed` to populate initial data if needed
+- Use I18n for translations
+
+## Migration Standards (Strong Migrations)
+
+### Naming Convention
+- Use descriptive names: `YYYYMMDDHHMMSS_descriptive_name.rb`
+
+### Safe Operations Examples
```

Comment:
maybe we can just tell it to follow strong_migration guides?

Reply:
I just find it strange to give strong_migrations so much importance (like 50 lines) in this foundational rule